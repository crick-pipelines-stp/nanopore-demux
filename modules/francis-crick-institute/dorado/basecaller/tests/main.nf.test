nextflow_process {

    name "Test Module DORADO_BASECALLER"
    script "../main.nf"
    process "DORADO_BASECALLER"

    tag "modules"
    tag "ont"
    tag "ont_dorado"
    tag "ont_dorado/basecaller"

    test("pod5") {
        tag "ont_dorado/basecaller/pod5"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/pod5/PAW36768_7b0e5258_08c7ccb8_0_sub100.pod5', checkIfExists: true)
                ])
                input[1] = []
                input[2] = 'hac'
                input[3] = []
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()}
            )
        }
    }

    test("pod5_bckit") {
        tag "ont_dorado/basecaller/pod5_bckit"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/pod5/PAW36768_7b0e5258_08c7ccb8_0_sub100.pod5', checkIfExists: true)
                ])
                input[1] = []
                input[2] = 'hac'
                input[3] = 'SQK-PCB111-24'
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()}
            )
        }
    }

    test("pod5_resume") {
        tag "ont_dorado/basecaller/pod5_resume"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test' ],
                    file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/pod5/PAW36768_7b0e5258_08c7ccb8_0_sub100.pod5', checkIfExists: true)
                ])
                input[1] = file('$baseDir/' + params.test_data_dir + '/ont/seq_runs/pod5_100/pod5.bam', checkIfExists: true)
                input[2] = 'hac'
                input[3] = []
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match()}
            )
        }
    }
}
