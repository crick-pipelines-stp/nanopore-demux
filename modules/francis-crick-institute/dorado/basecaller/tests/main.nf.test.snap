{
    "pod5_resume": {
        "content": [
            [
                "versions.yml:md5,ec84467cd957788bd357bfb86e31b871"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:31:07.830017"
    },
    "pod5": {
        "content": [
            [
                "versions.yml:md5,ec84467cd957788bd357bfb86e31b871"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:30:23.221196"
    },
    "pod5_bckit": {
        "content": [
            [
                "versions.yml:md5,ec84467cd957788bd357bfb86e31b871"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:30:45.450504"
    }
}